### PR TITLE
Add geometric random walk model to renewal session

### PR DESCRIPTION
**Pull request by seabbs**
I've been having some model and space problems in the nowcasting session so went back to the renewal session to investigate. 

I added in the random walk here as the normal based model isn't really working IMO and so feels like not a great way to end a session. Neither model are working well for me? @sbfnk did you have this working (i.e is it on my end) or is it something deeper?

_Note I added forecasting to this model so that we can use it later._

### CODE REVIEW

File: sessions/R-estimation-and-the-renewal-equation.qmd [hunk 1]
```diff
@@ -234,25 +235,18 @@ data <- list(
   ip_pmf = ip_pmf
 )
 r_inf_fit <- mod$sample(
-  data = data, refresh = 0, show_exceptions = FALSE, show_messages = FALSE
+  data = data, refresh = 0, show_exceptions = FALSE, show_messages = FALSE, parallel_chains = 4
```
Comment (sbfnk) [inline sessions/R-estimation-and-the-renewal-equation.qmd:238]: can we expect everyone will have >4 cores?

**Comment by sbfnk** [comment]
> I added in the random walk here as the normal based model isn't really working IMO and so feels like not a great way to end a session. Neither model are working well for me? @sbfnk did you have this working (i.e is it on my end) or is it something deeper?

I did but can also see now that it doesn't look good on the web site - suggesting that perhaps it could be fixed with suitable initial conditions?

**Comment by sbfnk** [comment]
Oh it was because of this (correct) change that it doesn't fit any more https://github.com/nfidd/nfidd/pull/108

File: sessions/R-estimation-and-the-renewal-equation.qmd (hunk 1 above)
Comment (seabbs) [reply sessions/R-estimation-and-the-renewal-equation.qmd:238]: this is just running each chain in parallel right? i.e max of 4 cores? but could drop it down to 2. I think if the models are working better probably don't need at all

**Comment by sbfnk** [comment]
> I added in the random walk here as the normal based model isn't really working IMO

Now that it works again we can consider moving this to the forecasting session as per the previous plan.

**Comment by seabbs** [comment]
> Now that iid normal works again we can consider moving the random walk to the forecasting session as per the previous plan.

The previous plan was for me to introduce it in the nowcasting session wasn't it? That session is absolutely full though. The more I look at this the more I feel uncomfortable with leaving the renewal session with such a bad model. If we can keep it here I think that might be good (really starts reinforcing the point about generative process as it will be the end of 3 sessions in a row.

**Comment by seabbs** [comment]
Somewhat forcing the matter the current version of the nowcasting session assumes this so I think we might need to merge and then address in a follow up issue (the nowcast session is currently mad so will need a refactor I think).

**Comment by sbfnk** [comment]
I think this makes sense. Let me do an initial review today/now and then we can merge / put outstanding things into separate issues.

File: stan/estimate-inf-and-r-rw.stan [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+generated quantities {
+  array[h] real forecast;
+  if (h > 0) {
+    for (i in 1:h) {
+      forecast[i] = poisson_rng(onsets[n + i]);
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (seabbs) [reply stan/estimate-inf-and-r-rw.stan:49]: so this was the tension I was wondering about. Do we really want to carry yet another duplicate model vs the cost of people seeing into the future? Maybe this is something to get feedback on?

[the remark this answers is not in the document]
Comment (sbfnk) [reply stan/estimate-inf-and-r-rw.stan:49]: Sounds good.

File: stan/functions/geometric_random_walk.stan [hunk 3]
```diff
@@ -0,0 +1,7 @@
+array[] real geometric_random_walk(real init, array[] real noise, real std) {
+    int n = num_elements(noise) + 1;
+    array[n] real x;
+    x[1] = init;
+    x[2:n] = to_array_1d(init + cumulative_sum(to_vector(noise) * std));
```
[the remark this answers is not in the document]
Comment (seabbs) [reply stan/functions/geometric_random_walk.stan:5]: agree but shall we move to an issue for this?

[the remark this answers is not in the document]
Comment (seabbs) [reply stan/functions/geometric_random_walk.stan:5]: If we want to write a R function to demonstrate this then perhaps these things can be done in the same issue?

**Pull request merged**